appmetrics/app_metrics.go

```diff
@@ -92,15 +92,17 @@ func (am *AppMetrics) updateCacheLoop() {
 			// update the database after closing the app map
 			// since this won't affect the app map, no need to continue touching it
 			am.db.Batch(func(tx *bolt.Tx) error {
+				b, err := tx.CreateBucketIfNotExists(am.appBucket)
+				if err != nil {
+					return fmt.Errorf("create bucket: %s", err)
+				}
+				// delete removed apps
 				for _, guid := range toRemove {
-					b, err := tx.CreateBucketIfNotExists(am.appBucket)
-					if err != nil {
-						return fmt.Errorf("create bucket: %s", err)
-					}
 					b.Delete([]byte(guid))
-					for guid, jsonApp := range updatedApps {
-						b.Put([]byte(guid), jsonApp)
-					}
+				}
```

Comment:
Was the old implementation overriding itself over and over?

Reply:
Yeah....

Reply:
And only if there were apps to delete in the first place